spec/controllers/course/survey/surveys_controller_spec.rb

```diff
@@ -7,14 +7,195 @@
+      end
+    end
+
+    describe '#show' do
+      let(:survey_traits) { :published }
+      let(:user) { student.user }
+
+      context 'when html page is requested' do
+        subject { get :show, course_id: course.id, id: survey.id }
+
+        it { is_expected.to render_template('index') }
+      end
+
+      context 'when html page is requested' do
+        subject { get :show, format: :json, course_id: course.id, id: survey.id }
+
+        it { is_expected.to render_template('_survey_with_questions') }
+      end
+    end
+
+    describe '#update' do
+      let(:user) { admin }
+      let(:survey_params) { {} }
+
+      subject do
+        patch :update, format: :json, course_id: course.id, id: survey.id,
+                       survey: survey_params
+      end
+
+      context 'when survey is anonymous' do
+        let(:survey_traits) { :anonymous }
+        let(:survey_params) { { anonymous: false } }
+
+        context 'when survey has no responses' do
+          before { subject }
+
+          it 'allows anonymous flag to be turned off' do
+            expect(survey.reload).not_to be_anonymous
+          end
+        end
+
+        context 'when survey has a student response' do
+          let!(:survey_response) do
+            create(:response, survey: survey, creator: student.user, course_user: student)
+          end
+
+          before do
+            survey_response
+            subject
+          end
+
+          it 'does not allow anonymous flag to be turned off' do
+            expect(survey.reload).to be_anonymous
+          end
+        end
+      end
+
+      context 'when updating fails' do
+        let(:survey_params) { { title: 'New Title' } }
+
+        before do
+          controller.instance_variable_set(:@survey, survey_stub)
+          subject
+        end
+
+        it { is_expected.to have_http_status(:bad_request) }
+      end
+    end
+
+    describe '#destroy' do
+      let(:user) { admin }
+      subject { delete :destroy, format: :json, course_id: course.id, id: survey.id }
+
+      context 'when destroy succeeds' do
+        it 'removes the deleted survey' do
+          expect { subject }.to change { course.surveys.count }.by(-1)
+          expect(response).to have_http_status(:ok)
+        end
+      end
+
+      context 'when destroy fails' do
+        before do
+          controller.instance_variable_set(:@survey, survey_stub)
+          subject
+        end
+
+        it { is_expected.to have_http_status(:bad_request) }
+      end
+    end
+
+    describe '#results' do
+      let(:user) { admin }
+
+      context 'when html page is requested' do
+        subject { get :results, course_id: course.id, id: survey.id }
+
+        it { is_expected.to render_template('index') }
+      end
+
+      context 'when json data is requested' do
+        subject { get :results, format: :json, course_id: course.id, id: survey.id }
+
+        it { is_expected.to render_template('results') }
```

Comment:
Updated to test the json data. `survey_with_question` is rendered by a few of the actions. Tested it in `show` only.